components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURL.java

```diff
@@ -84,11 +84,19 @@
 
     /**
      * Concatenate the protocol, host name, port, proxy context path, web context root, url context, query params and
-     * the fragment to return the absolute URL.
+     * the fragment to return the internal absolute URL.
      *
-     * @return The absolute URL from the Service URL instance.
+     * @return The internal absolute URL from the Service URL instance.
      */
-    String getAbsoluteURL();
+    String getAbsoluteURLInternal();
+
+    /**
```

Comment:
A description of how this URL differs from the internal URL would be useful for the API consumer.
Basically when to use this API over the API to get internal URL

Reply:
Shall we say getAbsoluteInternalURL

Reply:
+1